pylint/checkers/stdlib.py

```diff
@@ -446,6 +453,13 @@ class StdlibChecker(DeprecatedMixin, BaseChecker):
             "Calls to breakpoint(), sys.breakpointhook() and pdb.set_trace() should be removed "
             "from code that is not actively being debugged.",
         ),
+        "W1516": (
+            "lru_cache shouldn't be used on a method as it creates memory leaks",
+            "lru-cache-decorating-method",
+            "By decorating a method with lru_cache the 'self' argument will be linked to "
+            "to the lru_cache function and therefore never garbage collected. It is recommended "
+            "to refactor code to avoid this pattern.",
```

Comment:
```suggestion
            "to the lru_cache function and therefore never garbage collected. Unless your instance"
            "will never need to be garbage collected (singleton) it is recommended to refactor"
            "code to avoid this pattern.",
```